Maple2.Server.Game/Packets/FunctionCubePacket.cs

```diff
@@ -13,54 +15,62 @@ private enum Command : byte {
         Furniture = 5,
         SuccessLifeSkill = 8,
         FailLifeSkill = 9,
+        Feed = 11,
     }
 
     public static ByteWriter SendCubes(List<PlotCube> cubes) {
         var pWriter = Packet.Of(SendOp.FunctionCube);
         pWriter.Write<Command>(Command.SendCubes);
         pWriter.WriteInt(cubes.Count);
         foreach (PlotCube cube in cubes) {
-            pWriter.WriteUnicodeString(cube.InteractId);
-            pWriter.Write<InteractCubeState>(cube.InteractState);
-            pWriter.WriteByte(cube.InteractUnkByte);
+            pWriter.WriteClass<InteractCube>(cube.Interact!);
         }
         return pWriter;
     }
 
     public static ByteWriter AddFunctionCube(PlotCube cube) {
         var pWriter = Packet.Of(SendOp.FunctionCube);
         pWriter.Write<Command>(Command.Add);
-        pWriter.WriteUnicodeString(cube.InteractId);
-        pWriter.Write<InteractCubeState>(cube.InteractState);
-        pWriter.WriteByte(cube.InteractUnkByte);
+        pWriter.WriteClass<InteractCube>(cube.Interact!);
         return pWriter;
     }
 
     public static ByteWriter UseFurniture(long characterId, PlotCube cube) {
         var pWriter = Packet.Of(SendOp.FunctionCube);
         pWriter.Write<Command>(Command.Furniture);
         pWriter.WriteLong(characterId);
-        pWriter.WriteUnicodeString(cube.InteractId);
-        pWriter.WriteBool(cube.InteractState is InteractCubeState.InUse);
+        pWriter.WriteUnicodeString(cube.Interact!.Id);
+        pWriter.WriteBool(cube.Interact.State is InteractCubeState.InUse);
         return pWriter;
     }
 
     public static ByteWriter SuccessLifeSkill(long characterId, PlotCube cube) {
         var pWriter = Packet.Of(SendOp.FunctionCube);
         pWriter.Write<Command>(Command.SuccessLifeSkill);
         pWriter.WriteLong(characterId);
-        pWriter.WriteUnicodeString(cube.InteractId);
+        pWriter.WriteUnicodeString(cube.Interact!.Id);
         pWriter.WriteLong(DateTimeOffset.UtcNow.ToUnixTimeSeconds());
-        pWriter.Write<InteractCubeState>(cube.InteractState);
+        pWriter.Write<InteractCubeState>(cube.Interact.State);
         return pWriter;
     }
 
     public static ByteWriter FailLifeSkill(long characterId, PlotCube cube) {
         var pWriter = Packet.Of(SendOp.FunctionCube);
         pWriter.Write<Command>(Command.FailLifeSkill);
         pWriter.WriteLong(characterId);
-        pWriter.WriteUnicodeString(cube.InteractId);
+        pWriter.WriteUnicodeString(cube.Interact!.Id);
         pWriter.WriteLong(DateTimeOffset.UtcNow.ToUnixTimeSeconds());
         return pWriter;
     }
+
+    public static ByteWriter Feed(long itemUid, PlotCube cube) {
+        var pWriter = Packet.Of(SendOp.FunctionCube);
+        pWriter.Write<Command>(Command.Feed);
+        pWriter.WriteLong(itemUid);
+        pWriter.WriteUnicodeString(cube.Interact!.Id);
```

Comment:
..and as opposed to here, where we need to have the cube ID and all the InteractCube stuff. I feel like we should either just:
1. Make InteractCube non nullable and set a default (empty) InteractCube on all
2. Pass in the cube ID and InteractCube instead